src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -212,15 +224,22 @@
   (async/go (try
               (loop []
                 (let [task (async/<! lifecycle-chan)]
-                  (if (= task :shutdown)
-                    (do
-                      (log/debug "Received shutdown command, lifecycle worker exiting.")
-                      :done)
-                    (do
-                      (log/debug "Lifecycle worker executing lifecycle task.")
-                      (task)
-                      (log/debug "Lifecycle worker completed lifecycle task; awaiting next task.")
-                      (recur)))))
+                  (schema/validate LifeCycleTask task)
+                  (let [{:keys [type task-function]} task]
+                    (condp #(contains? %1 %2) type
+                      #{:shutdown}
+                      (do
+                        (log/debug "Received shutdown command, lifecycle worker exiting.")
+                        :done)
+
+                      #{:boot :restart}
+                      (do
+                        (log/debugf "Lifecycle worker executing %s lifecycle task." type)
+                        (task-function)
+                        (log/debugf "Lifecycle worker completed %s lifecycle task; awaiting next task." type)
+                        (recur))
+
+                      (throw (IllegalStateException. "Unrecognized lifecycle task:" task))))))
```

Comment:
Yeah, that makes sense. I guess I hope someone would catch this in a review, but there's not a guarantee of that, and it's probably better to guard against it since it isn't hard.